api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilderImpl.java

```diff
@@ -279,7 +289,22 @@ public HttpClient build() {
       throw new IllegalArgumentException(
           "No HTTP client factory for name '" + clientName + "' found");
     }
-    return httpClientFactory.buildClient(config);
+    HttpClient client = httpClientFactory.buildClient(config);
+
+    if (authentication != null) {
+      try {
+        authentication.start(cancellationCallbackConsumer);
```

Comment:
Also moved the call to `.start()` here

Reply:
Not that I'm against changing the `start()` method, but isn't this much simpler:

```
        authentication.start();
        cancellationCallbackConsumer.accept(authentication::close);
```